.github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js

```diff
@@ -1,13 +1,36 @@
 const core = require('@actions/core');
-const github = require('@actions/github');
+const {GitHub, getOctokitOptions} = require('@actions/github/lib/utils');
+const {throttling} = require('@octokit/plugin-throttling');
 const ActionUtils = require('../../libs/ActionUtils');
 const GithubUtils = require('../../libs/GithubUtils');
 
+const OctokitThrottled = GitHub.plugin(throttling);
+
 const prList = ActionUtils.getJSONInput('PR_LIST', {required: true});
 const isProd = ActionUtils.getJSONInput('IS_PRODUCTION_DEPLOY', {required: true});
 const version = core.getInput('DEPLOY_VERSION', {required: true});
 const token = core.getInput('GITHUB_TOKEN', {required: true});
-const octokit = github.getOctokit(token);
+const octokit = new OctokitThrottled(getOctokitOptions(token, {
```

Comment:
This looks like some pretty generic code to set up the throttling, so can we move it to the [`GithubUtils` constructor](https://github.com/Expensify/Expensify.cash/blob/3ea9446d5d6ef4762cb778e649395104184751ad/.github/libs/GithubUtils.js#L20-L22) ?

We could make `throttled` an optional parameter that sets up throttling in the constructor, or just do it by default. Thoughts? Is there any time we wouldn't want throttling when we hit a rate limit?

Reply:
Throttling is good for all cases. Should we retry after Abuse Limit? It will by default wait for the Abuse locking period.

Reply:
I guess for now we can not retry after the Abuse Limit – using throttling everywhere we shouldn't reach that limit unless something weird is going on.

Reply:
Agree.

Reply:
@roryabraham `GithubUtils` expects the first parameter as `octokit` instance but when we apply throttling, we have to create a new Octokit instance with the throttle plugin applied. 
```
const OctokitThrottled = GitHub.plugin(throttling);

const octokit = new OctokitThrottled(getOctokitOptions(token, {
```
Now how do you want me to handle this?
1. We Pass a second parameter `{throttle: Boolean}.` which ignores the first parameter's octokit instance and creates new with the above logic.
2. We check for throttle property in the first parameter and if this property is found we Assume that we have passed the options as the first parameter and we will create our own Instance and if this property does not exist we assume that the first param was `octokit` instance so do whatever we do now.

Reply:
I guess what I would do it just get rid of the `octokit` parameter in the `GithubUtils` constructor and just set it up like you have here, directly in the constructor when someone creates a `GithubUtils` instance. Then in all the actions that call `new GithubUtils()` just get rid of the `octokit` parameter. Then as a bonus step you could make `GithubUtils` a singleton such that all the methods become static and internally it will initialize the `octokit` instance if it needs to.

However, the only thing I'm unsure of in this approach is that when you do this:

```javascript
const core = require('@actions/core');
const github = require('@actions/github');
const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
```

You get a pre-authenticated `octokit` client. I'm not sure if that happens the same using this:

```javascript
const {GitHub, getOctokitOptions} = require('@actions/github/lib/utils');
```

because you're not providing that `GITHUB_TOKEN` which is provided from the GH workflow that's calling the action.

Reply:
We pass the token here `const octokit = new OctokitThrottled(getOctokitOptions(token <----, {` I checked `github.getOctokit(` source. It basically does the the same.

Reply:
Oh, nice 👍 

So yeah I'd recommend that you just instantiate the `octokit` instance in the `GithubUtils` constructor. It's up to you if you want to make the class a singleton